### PR TITLE
bitrise-step-ionic-build 0.0.1

### PR DESCRIPTION
**Pull request by non-binary**
TagCheck

### What to do if the build fails?

At the moment contributors do not have access to the CI workflow triggered by StepLib PRs. In case of a failed build, we ask for your patience. Maintainers of Bitrise Steplib will sort it out for you or inform you if any further action is needed.

### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [x] __I will not move an already shared step version's tag to another commit__
- [x] I read the [Step Development Guideline](https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md)
- [x] I have a test for my Step, which can be run with `bitrise run test` (in the step's repository)
- [x] I did run `bitrise run audit-this-step` (in the step's repository - note, if you don't have this workflow in your `bitrise.yml`, [you can copy it from the step template](https://github.com/bitrise-steplib/step-template/blob/master/bitrise.yml).)
- [x] I read and accept the [Abandoned Step policy](https://github.com/bitrise-io/bitrise-steplib#abandoned-step-policy)

**New Step**
Thank you for the new Step share! The CI check might will fail due to our extended validation engine. Nothing to worry about yet, we will get back to you shortly.

### CODE REVIEW

**Comment by non-binary** [comment]
The build failed with

```
Invalid spec
/steps/bitrise-step-ionic-build/info: {} "maintainer" value is required
/steps/bitrise-step-ionic-build/versions/0.0.1/deps/check_only/0/name: "ionic" regexp pattrn ^xcode$|^gradle$ mismatch on string: ionic
```

I couldn't find where the maintainer field is. Where do I need to input that?

Secondly, I don't fully understand the ionic error as it seems the regex pattern wants either xcode or gradle but I would like to check if Ionic is installed and this works locally.

Any ideas?

**Comment by non-binary** [comment]
@trapacska @lpusok Can you take a look at this please?

File: steps/bitrise-step-ionic-build/0.0.1/step.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      Set to yes if you want `--prod` to be used.
+    is_expand: true
+    is_required: true
+    summary: To use the production configuration.
+    title: Production Configuration
+    value_options:
+    - "true"
+    - "false"
+  production_mode: "no"
+- advanced_options: ""
```
Comment (lpusok) [inline steps/bitrise-step-ionic-build/0.0.1/step.yml:39]: Could call it just "options" to be consistent with ionic-archive step, but no huge objections.

File: steps/bitrise-step-ionic-build/0.0.1/step.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+inputs:
+- opts:
+    description: |
+      Set to yes if you want `--prod` to be used.
+    is_expand: true
+    is_required: true
+    summary: To use the production configuration.
+    title: Production Configuration
+    value_options:
+    - "true"
```
[the remark this answers is not in the document]
Comment (non-binary) [reply steps/bitrise-step-ionic-build/0.0.1/step.yml:36]: Fixed. Thanks!

File: steps/bitrise-step-ionic-build/0.0.1/step.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  The Step runs ionic build with the arguments you provide.
+description: |
+  The Step runs ionic build with the arguments you provide.
+website: https://github.com/non-binary/bitrise-step-ionic-build
+source_code_url: https://github.com/non-binary/bitrise-step-ionic-build
+support_url: https://github.com/non-binary/bitrise-step-ionic-build/issues
+published_at: 2020-06-04T14:00:54.594696-04:00
+source:
+  git: https://github.com/non-binary/bitrise-step-ionic-build.git
+  commit: 0d5664d634c89c2c3dd52cd24f09615ba3575921
```
[the remark this answers is not in the document]
Comment (non-binary) [reply steps/bitrise-step-ionic-build/0.0.1/step.yml]: Okay, on it!

File: steps/bitrise-step-ionic-build/0.0.1/step.yml (hunk 1 above)
Comment (non-binary) [reply steps/bitrise-step-ionic-build/0.0.1/step.yml]: Fixed. Thanks!

**Comment by non-binary** [comment]
@lpusok @trapacska Ready for merging, I believe. Feel free to give me any other feedback if you have any. :) Thank you!

**Pull request merged**